File: chapters/chapter5/scripts/matlab/u_D.m
function f=u_D(x1,x2,sdl) 
  f=zeros(size(x1)) ;
  % segments in local numeration:
  % 1-st segment 
  I=find(sdl==1); f(I)=x1(I).^2+x2(I).^2;
  % 2-nd segment 
  I=find(sdl==2); f(I)=x1(I).^2+x2(I).^2 ;
end

File: chapters/appendices/scripts/python/uD.py
def uD(x1,x2,sdl): 
    f=numpy.zeros(x1.shape)
    # segments in local numeration:
    # 1-st segment 
    I=sdl==0
    f[I]=numpy.square(x1[I])+numpy.square(x2[I])
    # 2-nd segment 
    I=sdl==1
    f[I]=numpy.square(x1[I])+numpy.square(x2[I])
    return f